### PR TITLE
pgBackRest Compaction into crunchy-pgbackrest(-repo)

### PR DESCRIPTION
**Pull request by jmckulk**
This update compacts the functionality of the existing
crunchy-backrest-restore container along with the
pgo-backrest and pgo-backrest-repo containers from the
Crunchy Postgres Operator project into a main container,
crunchy-pgbackrest, as well as a subsequent container,
crunchy-pgbackrest-repo, that allows for a separate user
as currently used by the existing pgBackRest implementation.

The various pgBackRest operating functions from the existing
containers are now accomplished by the use of operating modes.
These modes are as follows:

pgbackrest mode is used for taking pgBackRest backups, retrieving
info and stanza creation.

pgbackrest-repo mode acts as a pgBackRest remote repository for
the Postgres cluster to use for storing archive files and backups.

pgbackrest-restore mode executes a pgBackRest restore independent
of the Crunchy PostgreSQL Operator.

Finally, as mentioned above, the pgbackrest-repo container allows
for the operation in the 'pgbackrest-repo' mode, but with a
separate, pgbackrest user.

The final container incorporates all necessary scripts, configurations
and code necessary to execute the various modes, while removing any
redundant or otherwise unnecessary dependencies. A new script,
start.sh, determines the correct operating mode based on the value
in the MODE environment variable, and then calls the correct subset
of the overall functionality.

**Checklist:**

 
 - [x] Have you added an explanation of what your changes do and why you'd like them to be included?
 - [x] Have you updated or added documentation for the change, as applicable?
 - [x] Have you tested your changes on all related environments with successful results, as applicable?

**Type of Changes:**

 
 - [ ] Bug fix (non-breaking change which fixes an issue)
 - [x] New feature (non-breaking change which adds functionality)
 - [x] Breaking change (fix or feature that would cause existing functionality to change)

**What is the current behavior? (link to any open issues here)**

**What is the new behavior (if this is a feature change)?**

**Other information**:

### CODE REVIEW

File: bin/pgbackrest-repo/pgbackrest-repo.sh [hunk 1]
```diff
@@ -0,0 +1,81 @@
...
+	if [[ "${PGBACKREST_REPO1_S3_URI_STYLE}" != "" ]]
+	then
+		printf "repo1-s3-uri-style=%s\n" "${PGBACKREST_REPO1_S3_URI_STYLE}" >> /etc/pgbackrest/pgbackrest.conf
+	fi
+	
+fi
+
+mkdir -p ~/.ssh/
+cp $CONFIG/config ~/.ssh/
+#cp $CONFIG/authorized_keys ~/.ssh/
```
Comment (tjmoore4) [inline bin/pgbackrest-repo/pgbackrest-repo.sh]: looks like we missed this, should be able to remove that comment.

File: bin/pgbackrest-restore/restore_common_lib.sh [hunk 2]
```diff
@@ -13,9 +13,11 @@
...
 # limitations under the License.
 
-export LD_PRELOAD=/usr/lib64/libnss_wrapper.so
-export NSS_WRAPPER_PASSWD=/tmp/passwd
-export NSS_WRAPPER_GROUP=/tmp/group
+CRUNCHY_DIR=${CRUNCHY_DIR:-'/opt/crunchy'}
+
+#export LD_PRELOAD=/usr/lib64/libnss_wrapper.so
+#export NSS_WRAPPER_PASSWD=/tmp/passwd
+#export NSS_WRAPPER_GROUP=/tmp/group
```
Comment (tjmoore4) [inline bin/pgbackrest-restore/restore_common_lib.sh]: should be able to remove these comments as well.

File: cmd/pgbackrest/main.go [hunk 3]
```diff
@@ -0,0 +1,237 @@
...
+	// explicitly set
+	S3VerifyTLS, _ := strconv.ParseBool(os.Getenv("PGHA_PGBACKREST_S3_VERIFY_TLS"))
+	log.Debugf("setting PGHA_PGBACKREST_S3_VERIFY_TLS to %v", S3VerifyTLS)
+
+	/*
+		client, err := kubeapi.NewClient()
+		if err != nil {
+			panic(err)
+		}
+	*/
```
Comment (tjmoore4) [inline cmd/pgbackrest/main.go]: not sure why this was here, but looks like it can be removed now?

File: CONTRIBUTING.md [hunk 4]
```diff
@@ -118,7 +118,7 @@ for what you're working on.
 Overall, release tags attempt to follow the
 [semantic versioning](https://semver.org) scheme. The Crunchy Container Suite is
 closely related to the
-[PostgreSQL Operator](https://github.com/CrunchyData/postgres-operator) and in
+[PostgreSQL Operator](https://github.com/CrunchyData/crunchy-containers) and in
```
Comment (jkatz) [inline CONTRIBUTING.md]: This was correct the way it was before.

File: bin/pgbackrest-repo/archive-push-s3.sh [hunk 5]
```diff
@@ -0,0 +1,3 @@
+#!/bin/bash
```
Comment (jkatz) [inline bin/pgbackrest-repo/archive-push-s3.sh:1]: This needs the license

File: bin/pgbackrest/README.txt [hunk 6]
```diff
@@ -0,0 +1,3 @@
+pgbackrest binary goes in this directory and gets
+copied into the crunchy-pgbackrest image, .gitignore is here
+to keep the binary from making its way into github
```
Comment (jkatz) [inline bin/pgbackrest/README.txt]: This file feels useless, I'd just remove it. the `.gitignore` satisfies this

File: bin/uid_postgres.sh [hunk 7]
```diff
@@ -0,0 +1,22 @@
+#!/bin/bash
```
Comment (jkatz) [inline bin/uid_postgres.sh]: Needs the license here...but also curious why this is in this patch and not with the original compaction

**Review by jkatz — CHANGES_REQUESTED** [review]
Good start, seems to be going in the right direction. A few pieces of feedback.

File: bin/uid_postgres.sh (hunk 7 above)
Comment (jmckulk) [reply bin/uid_postgres.sh]: Thats a good question. Ill look into it why and get it fixed

**Review by jkatz — APPROVED** [review]

**Pull request merged**